prometheus/base/prometheus.rules.yaml

```diff
@@ -1,13 +1,12 @@
-apiVersion: monitoring.coreos.com/v1
-kind: PrometheusRule
-metadata:
-  labels:
-    openshift.io/prometheus-rule-evaluation-scope: leaf-prometheus
-  name: o11y-example-rule
-  namespace: o11y
-spec:
-  groups:
-    - name: example
-      rules:
-        - record: o11y_prometheus_http_requests_total
-          expr: sum by (code) (prometheus_http_requests_total)
+
+groups:
+  - name: stonesoup-metering
+    interval: 30s
+    rules:
+       - record: container:network_transmit_bytes:group
+         expr: |2
+            container_network_transmit_bytes_total
+            * on (namespace, pod) 
```

Comment:
does that include init containers?

Reply:
I didn't see any metric or query to measure network in init containers.  
I will keep on looking...

Reply:
Seems like it really does not exist. 
If we can expect some of the init_container name (e.g: it starts with `init_`)
we can try filtering with the container name... 
although that's just a filter on the query we already use, so I'm not sure why it does not apply for init_containers.

Reply:
Is there network activity in init containers?

Reply:
AFAIK anything can happen in an init container, so it depends...

Reply:
If we want to monitor network traffic from init containers as well, shouldn't the `container` label be added to the `on` clause?
For example:
`* on (namespace, pod, container) `

Reply:
The only container that appears in the "core" metric `container_network_transmit_bytes_total` is the `POD` container, so adding it will not solve this.

I cannot find any init_container related metrics other the the resource limits

Reply:
upon checking, I think we can assume for now there is no egress usage from init containers. 
even when the query is not filtered, we cannot get egress info, so filters won't help here, and there is no specific query for the init containers egress. This is something that can be solved (if needed) only with extreme measures we've discussed and set aside for the current solution, such as Ebpf.

Reply:
I will test it today by creating a pod with container and init container, both will run some network egress, and see if I can get the expected total egress

Reply:
As we saw in our tests, only the first `init container` (if defined) would be detected, no other init containers or containers would be detected afterwards.
This is coming from `cAdvisor`...

Reply:
Let's open a ticket for that so we try to address it later on.